packages/backend/src/ee/services/ai/tools/findExplores.ts

```diff
@@ -15,104 +10,111 @@ type Dependencies = {
+                  })
+                : table.description
+        }
     </Description>
 
     ${
         table.joinedTables && table.joinedTables.length > 0
             ? `
-    <JoinedTables alt="IDs of the joined tables" size="${
+    <JoinedTables alt="IDs of the joined tables" totalCount="${
         table.joinedTables.length
     }">
         ${table.joinedTables
-            .map(
-                (joinedTable) =>
-                    `<JoinedTable id="${joinedTable}">${joinedTable}</JoinedTable>`,
-            )
-            .join('\n\n')}
+            .map((joinedTable) => `<Table>${joinedTable}</Table>`)
+            .join('\n        ')}
     </JoinedTables>
 `.trim()
             : ''
     }
 
-    <Fields alt="All fields from all tables" size="${
-        dimensions.length + metrics.length
-    }">
-        <Dimensions alt="dimension ids" size="${dimensions.length}">
+    <Fields alt="All fields from all tables" totalCount="${
+        (dimensionsPagination?.totalResults ?? 0) +
+        (metricsPagination?.totalResults ?? 0)
+    }" displayedResults="${dimensions.length + metrics.length}">
+        <Dimensions alt="most popular dimensions" totalResults="${
+            dimensionsPagination?.totalResults ?? 0
+        }" displayedResults="${dimensions.length}">
             ${dimensions
                 .map(
                     (d) =>
-                        `<Dimension fieldId="${getItemId({
+                        `<Dimension table="${d.tableName}" name="${
+                            d.name
+                        }" fieldId="${getItemId({
                             name: d.name,
                             table: d.tableName,
-                        })}" table="${d.tableName}" name="${d.name}" label="${
-                            d.label
-                        }" />`,
+                        })}">${d.label}</Dimension>`,
                 )
-                .join('\n')}
+                .join('\n            ')}
         </Dimensions>
 
-        <Metrics alt="metric ids sorted by popularity" size="${metrics.length}">
+        <Metrics alt="most popular metrics" totalResults="${
+            metricsPagination?.totalResults ?? 0
+        }" displayedResults="${metrics.length}">
             ${metrics
                 .map(
                     (m) =>
-                        `<Metric fieldId="${getItemId({
+                        `<Metric table="${m.tableName}" name="${
+                            m.name
+                        }" fieldId="${getItemId({
                             name: m.name,
                             table: m.tableName,
-                        })}" table="${m.tableName}" name="${m.name}" label="${
-                            m.label
-                        }" />`,
+                        })}">${m.label}</Metric>`,
                 )
-                .join('\n')}
+                .join('\n            ')}
         </Metrics>
     </Fields>
 </Explore>`.trim();
```

Comment:
note: jsx to xml renderer would be nice here

Reply:
💯 agree